### ISSUE TITLE
Async Arrow Function "Unexpected token ="

### ISSUE THREAD

**Issue by Biuni** [eslint/eslint#10127]
**Tell us about your environment**

* **ESLint Version:** 4.18.1
* **Node Version:** 8.9.1
* **npm Version:** 5.5.1

**What parser (default, Babel-ESLint, etc.) are you using?**
Babel-ESlint

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
  "extends": "airbnb",
  "parserOptions": {
    "ecmaVersion": 8
  },
  "env": {
    "jest": true,
    "browser": true
  },
  "rules": {
    "semi": [
      2,
      "never"
    ],
    "no-unexpected-multiline": 2,
    "import/no-named-as-default": 0,
    "import/no-named-as-default-member": 0,
    "react/prop-types": 0,
    "react/jsx-filename-extension": [
      1,
      {
        "extensions": [
          ".js",
          ".jsx"
        ]
      }
    ],
    "jsx-a11y/anchor-is-valid": [
      "error",
      {
        "components": [
          "Link"
        ],
        "specialLink": [
          "to"
        ]
      }
    ],
    "no-unused-expressions": [
      "error",
      {
        "allowTernary": true
      }
    ]
  }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
  callApi = async (url) => {
    const response = await fetch(url)
    const body = await response.json()

    if (response.status !== 200) {
      throw Error(body.message)
    }

    return body
  }
```

```bash
eslint src
```

**What did you expect to happen?**
ESLint done.

**What actually happened? Please include the actual, raw output from ESLint.**
" error  Parsing error: Unexpected token = " 
[ callApi = async (url) .... is the problem ]

**Comment by platinumazure** [comment]
It seems like everything should work, assuming airbnb has set everything up correctly in their config file.

Can you try running ESLint with the `--print-config` flag (passing in a filename to lint), as well as with the `--debug` flag? Using one or both of those, you may be able to see if there is an unexpected configuration being pulled in.

**Comment by Biuni** [comment]
I run `eslint src --debug` and print this error: `Cannot find module 'eslint-config-airbnb'`.
Could it be that i'm using global eslint instead of local eslint?

**Comment by platinumazure** [comment]
Could be. Are you running ESLint from an npm script normally? If so, you might need to run `./node_modules/.bin/eslint` instead of `eslint`.

**Comment by Biuni** [comment]
Yes, it's a simply npm script. Now i've tried `./node_modules/.bin/eslint src --debug` and the module 'eslint-config-airbnb' it's ok. However the problem of "Unexpected token" it's the same.

**Comment by platinumazure** [comment]
Thanks! Could you please share the full output of running with the --debug
flag? (Also, take a look to see if any other config files are being loaded
that you don't expect)

**Comment by Biuni** [comment]
This's the output. 
```
eslint:cli Running on files +0ms
eslint:glob-util Creating list of files to process. +0ms
eslint:ignored-paths Looking for ignore file in D:\Javascript\dev-resume +0ms
eslint:ignored-paths Could not find ignore file in cwd +1ms
eslint:cli-engine Processing D:\Javascript\devTest\src\App.js +0ms
eslint:cli-engine Linting D:\Javascript\devTest\src\App.js +2ms
eslint:config Constructing config file hierarchy for D:\Javascript\devTest\src +0ms
eslint:config Using .eslintrc and package.json files +0ms
eslint:config Loading D:\Javascript\devTest\.eslintrc +3ms
eslint:config-file Loading config file: D:\Javascript\devTest\.eslintrc +0ms
eslint:config-file Loading airbnb +136ms
eslint:config-file Attempting to resolve eslint-config-airbnb +1ms
eslint:config-file Loading JS config file: D:\Javascript\devTest\node_modules\eslint-config-airbnb\index.js +2ms
eslint:config-file Loading D:\Javascript\devTest\node_modules\eslint-config-airbnb\rules\react-a11y.js +5ms
eslint:config-file Loading JS config file: D:\Javascript\devTest\node_modules\eslint-config-airbnb\rules\react-a11y.js +1ms
eslint:config-file Loading D:\Javascript\devTest\node_modules\eslint-config-airbnb\rules\react.js +690ms
eslint:config-file Loading JS config file: D:\Javascript\devTest\node_modules\eslint-config-airbnb\rules\react.js +1ms
eslint:config-file Loading D:\Javascript\devTest\node_modules\eslint-config-airbnb-base\rules\strict.js +53ms
eslint:config-file Loading JS config file: D:\Javascript\devTest\node_modules\eslint-config-airbnb-base\rules\strict.js +1ms
eslint:config-file Loading D:\Javascript\devTest\node_modules\eslint-config-airbnb-base\index.js +6ms
eslint:config-file Loading JS config file: D:\Javascript\devTest\node_modules\eslint-config-airbnb-base\index.js +1ms
eslint:config-file Loading D:\Javascript\devTest\node_modules\eslint-config-airbnb-base\rules\imports.js +2ms
eslint:config-file Loading JS config file: D:\Javascript\devTest\node_modules\eslint-config-airbnb-base\rules\imports.js +1ms
eslint:config-file Loading D:\Javascript\devTest\node_modules\eslint-config-airbnb-base\rules\es6.js +297ms
eslint:config-file Loading JS config file: D:\Javascript\devTest\node_modules\eslint-config-airbnb-base\rules\es6.js +1ms
eslint:config-file Loading D:\Javascript\devTest\node_modules\eslint-config-airbnb-base\rules\variables.js +68ms
eslint:config-file Loading JS config file: D:\Javascript\devTest\node_modules\eslint-config-airbnb-base\rules\variables.js +1ms
eslint:config-file Loading D:\Javascript\devTest\node_modules\eslint-config-airbnb-base\rules\style.js +23ms
eslint:config-file Loading JS config file: D:\Javascript\devTest\node_modules\eslint-config-airbnb-base\rules\style.js +0ms
eslint:config-file Loading D:\Javascript\devTest\node_modules\eslint-config-airbnb-base\rules\node.js +227ms
eslint:config-file Loading JS config file: D:\Javascript\devTest\node_modules\eslint-config-airbnb-base\rules\node.js +1ms
eslint:config-file Loading D:\Javascript\devTest\node_modules\eslint-config-airbnb-base\rules\errors.js +6ms
eslint:config-file Loading JS config file: D:\Javascript\devTest\node_modules\eslint-config-airbnb-base\rules\errors.js +1ms
eslint:config-file Loading D:\Javascript\devTest\node_modules\eslint-config-airbnb-base\rules\best-practices.js +61ms
eslint:config-file Loading JS config file: D:\Javascript\devTest\node_modules\eslint-config-airbnb-base\rules\best-practices.js +1ms
eslint:config Using D:\Javascript\devTest\.eslintrc +2s
eslint:config-ops Using config from partial cache +0ms
eslint:config-ops Apply environment settings to config +1ms
eslint:config-ops Creating config for environment node +1ms
eslint:config-ops Creating config for environment es6 +1ms
eslint:config-ops Creating config for environment jest +0ms
eslint:config-ops Creating config for environment browser +1ms
eslint:linter Linting code for D:\Javascript\devTest\src\App.js (pass 1) +0ms
eslint:linter Generating fixed text for D:\Javascript\devTest\src\App.js (pass 1) +223ms
eslint:text-fixer Applying fixes +0ms
eslint:text-fixer shouldFix parameter was false, not attempting fixes +0ms
eslint:cli-engine Processing D:\Javascript\devTest\src\component\page\Ama.js +2s
eslint:cli-engine Linting D:\Javascript\devTest\src\component\page\Ama.js +0ms
eslint:config Constructing config file hierarchy for D:\Javascript\devTest\src\component\page +233ms
eslint:config Using .eslintrc and package.json files +0ms
eslint:config Using config from cache +2ms
eslint:linter Linting code for D:\Javascript\devTest\src\component\page\Ama.js (pass 1) +5ms
eslint:linter Generating fixed text for D:\Javascript\devTest\src\component\page\Ama.js (pass 1) +6ms
eslint:text-fixer Applying fixes +11ms
eslint:text-fixer shouldFix parameter was false, not attempting fixes +0ms
eslint:cli-engine Processing D:\Javascript\devTest\src\component\page\Discussions.js +11ms
eslint:cli-engine Linting D:\Javascript\devTest\src\component\page\Discussions.js +1ms
eslint:config Constructing config file hierarchy for D:\Javascript\devTest\src\component\page +9ms
eslint:config Using .eslintrc and package.json files +1ms
eslint:config Using config from cache +1ms
eslint:linter Linting code for D:\Javascript\devTest\src\component\page\Discussions.js (pass 1) +5ms
eslint:linter Generating fixed text for D:\Javascript\devTest\src\component\page\Discussions.js (pass 1) +6ms
eslint:text-fixer Applying fixes +10ms
eslint:text-fixer shouldFix parameter was false, not attempting fixes +1ms
eslint:cli-engine Processing D:\Javascript\devTest\src\component\page\Home.js +10ms
eslint:cli-engine Linting D:\Javascript\devTest\src\component\page\Home.js +1ms
eslint:config Constructing config file hierarchy for D:\Javascript\devTest\src\component\page +9ms
eslint:config Using .eslintrc and package.json files +1ms
eslint:config Using config from cache +1ms
eslint:linter Linting code for D:\Javascript\devTest\src\component\page\Home.js (pass 1) +5ms
eslint:linter Generating fixed text for D:\Javascript\devTest\src\component\page\Home.js (pass 1) +81ms
eslint:text-fixer Applying fixes +86ms
eslint:text-fixer shouldFix parameter was false, not attempting fixes +0ms
eslint:cli-engine Processing D:\Javascript\devTest\src\component\page\NotFound.js +85ms
eslint:cli-engine Linting D:\Javascript\devTest\src\component\page\NotFound.js +0ms
eslint:config Constructing config file hierarchy for D:\Javascript\devTest\src\component\page +84ms
eslint:config Using .eslintrc and package.json files +0ms
eslint:config Using config from cache +1ms
eslint:linter Linting code for D:\Javascript\devTest\src\component\page\NotFound.js (pass 1) +4ms
eslint:linter Generating fixed text for D:\Javascript\devTest\src\component\page\NotFound.js (pass 1) +26ms
eslint:text-fixer Applying fixes +30ms
eslint:text-fixer shouldFix parameter was false, not attempting fixes +1ms
eslint:cli-engine Processing D:\Javascript\devTest\src\component\page\Resume.js +31ms
eslint:cli-engine Linting D:\Javascript\devTest\src\component\page\Resume.js +1ms
eslint:config Constructing config file hierarchy for D:\Javascript\devTest\src\component\page +31ms
eslint:config Using .eslintrc and package.json files +1ms
eslint:config Using config from cache +1ms
eslint:linter Linting code for D:\Javascript\devTest\src\component\page\Resume.js (pass 1) +8ms
eslint:linter Generating fixed text for D:\Javascript\devTest\src\component\page\Resume.js (pass 1) +30ms
eslint:text-fixer Applying fixes +36ms
eslint:text-fixer shouldFix parameter was false, not attempting fixes +1ms
eslint:cli-engine Processing D:\Javascript\devTest\src\component\shared\Article.js +35ms
eslint:cli-engine Linting D:\Javascript\devTest\src\component\shared\Article.js +2ms
eslint:config Constructing config file hierarchy for D:\Javascript\devTest\src\component\shared +34ms
eslint:config Using .eslintrc and package.json files +1ms
eslint:config Using config from cache +1ms
eslint:linter Linting code for D:\Javascript\devTest\src\component\shared\Article.js (pass 1) +8ms
eslint:linter Generating fixed text for D:\Javascript\devTest\src\component\shared\Article.js (pass 1) +119ms
eslint:text-fixer Applying fixes +127ms
eslint:text-fixer shouldFix parameter was false, not attempting fixes +1ms
eslint:cli-engine Processing D:\Javascript\devTest\src\component\shared\Placeholder.js +126ms
eslint:cli-engine Linting D:\Javascript\devTest\src\component\shared\Placeholder.js +1ms
eslint:config Constructing config file hierarchy for D:\Javascript\devTest\src\component\shared +125ms
eslint:config Using .eslintrc and package.json files +1ms
eslint:config Using config from cache +1ms
eslint:linter Linting code for D:\Javascript\devTest\src\component\shared\Placeholder.js (pass 1) +5ms
eslint:linter Generating fixed text for D:\Javascript\devTest\src\component\shared\Placeholder.js (pass 1) +54ms
eslint:text-fixer Applying fixes +63ms
eslint:text-fixer shouldFix parameter was false, not attempting fixes +1ms
eslint:cli-engine Processing D:\Javascript\devTest\src\component\shared\QeA.js +64ms
eslint:cli-engine Linting D:\Javascript\devTest\src\component\shared\QeA.js +0ms
eslint:config Constructing config file hierarchy for D:\Javascript\devTest\src\component\shared +63ms
eslint:config Using .eslintrc and package.json files +1ms
eslint:config Using config from cache +1ms
eslint:linter Linting code for D:\Javascript\devTest\src\component\shared\QeA.js (pass 1) +12ms
eslint:linter Generating fixed text for D:\Javascript\devTest\src\component\shared\QeA.js (pass 1) +30ms
eslint:text-fixer Applying fixes +36ms
eslint:text-fixer shouldFix parameter was false, not attempting fixes +1ms
eslint:cli-engine Processing D:\Javascript\devTest\src\component\shared\Sidebar.js +36ms
eslint:cli-engine Linting D:\Javascript\devTest\src\component\shared\Sidebar.js +1ms
eslint:config Constructing config file hierarchy for D:\Javascript\devTest\src\component\shared +34ms
eslint:config Using .eslintrc and package.json files +1ms
eslint:config Using config from cache +1ms
eslint:linter Linting code for D:\Javascript\devTest\src\component\shared\Sidebar.js (pass 1) +5ms
eslint:linter Generating fixed text for D:\Javascript\devTest\src\component\shared\Sidebar.js (pass 1) +118ms
eslint:text-fixer Applying fixes +122ms
eslint:text-fixer shouldFix parameter was false, not attempting fixes +1ms
eslint:cli-engine Processing D:\Javascript\devTest\src\global.js +123ms
eslint:cli-engine Linting D:\Javascript\devTest\src\global.js +1ms
eslint:config Constructing config file hierarchy for D:\Javascript\devTest\src +122ms
eslint:config Using .eslintrc and package.json files +0ms
eslint:config Using config from cache +1ms
eslint:linter Linting code for D:\Javascript\devTest\src\global.js (pass 1) +6ms
eslint:linter Generating fixed text for D:\Javascript\devTest\src\global.js (pass 1) +19ms
eslint:text-fixer Applying fixes +24ms
eslint:text-fixer shouldFix parameter was false, not attempting fixes +1ms
eslint:cli-engine Processing D:\Javascript\devTest\src\index.js +23ms
eslint:cli-engine Linting D:\Javascript\devTest\src\index.js +1ms
eslint:config Constructing config file hierarchy for D:\Javascript\devTest\src +23ms
eslint:config Using .eslintrc and package.json files +1ms
eslint:config Using config from cache +1ms
eslint:linter Linting code for D:\Javascript\devTest\src\index.js (pass 1) +5ms
eslint:linter Generating fixed text for D:\Javascript\devTest\src\index.js (pass 1) +41ms
eslint:text-fixer Applying fixes +46ms
eslint:text-fixer shouldFix parameter was false, not attempting fixes +0ms
eslint:cli-engine Processing D:\Javascript\devTest\src\registerServiceWorker.js +46ms
eslint:cli-engine Linting D:\Javascript\devTest\src\registerServiceWorker.js +1ms
eslint:config Constructing config file hierarchy for D:\Javascript\devTest\src +45ms
eslint:config Using .eslintrc and package.json files +1ms
eslint:config Using config from cache +1ms
eslint:linter Linting code for D:\Javascript\devTest\src\registerServiceWorker.js (pass 1) +7ms
eslint:linter Generating fixed text for D:\Javascript\devTest\src\registerServiceWorker.js (pass 1) +85ms
eslint:text-fixer Applying fixes +92ms
eslint:text-fixer shouldFix parameter was false, not attempting fixes +0ms
eslint:cli-engine Processing D:\Javascript\devTest\src\routing.js +91ms
eslint:cli-engine Linting D:\Javascript\devTest\src\routing.js +0ms
eslint:config Constructing config file hierarchy for D:\Javascript\devTest\src +90ms
eslint:config Using .eslintrc and package.json files +0ms
eslint:config Using config from cache +1ms
eslint:linter Linting code for D:\Javascript\devTest\src\routing.js (pass 1) +5ms
eslint:linter Generating fixed text for D:\Javascript\devTest\src\routing.js (pass 1) +16ms
eslint:text-fixer Applying fixes +20ms
eslint:text-fixer shouldFix parameter was false, not attempting fixes +0ms
eslint:cli-engine Linting complete in: 2679ms +20ms
```

**Comment by codeblooded** [comment]
I'm facing this exact same problem:

```
users.routes.test.js
  6:55  error  Parsing error: Unexpected token =>
```

and my code:

```javascript
    it('fails if no contact email provided', async () => {
      const res = await req(app).post('/users').send({
        firstName: "bendi",
        lastName: "testName",
        username: "asdfasdf",
        password: "Dask4fffasf"
      });
      expect(res.statusCode).toBe(400);
    });
```

Fails on the `async () => {` and this is my config:

```
{
  "globals": {
    "Array": false,
    "ArrayBuffer": false,
    "Boolean": false,
    "constructor": false,
    "DataView": false,
    "Date": false,
    "decodeURI": false,
    "decodeURIComponent": false,
    "encodeURI": false,
    "encodeURIComponent": false,
    "Error": false,
    "escape": false,
    "eval": false,
    "EvalError": false,
    "Float32Array": false,
    "Float64Array": false,
    "Function": false,
    "hasOwnProperty": false,
    "Infinity": false,
    "Int16Array": false,
    "Int32Array": false,
    "Int8Array": false,
    "isFinite": false,
    "isNaN": false,
    "isPrototypeOf": false,
    "JSON": false,
    "Map": false,
    "Math": false,
    "NaN": false,
    "Number": false,
    "Object": false,
    "parseFloat": false,
    "parseInt": false,
    "Promise": false,
    "propertyIsEnumerable": false,
    "Proxy": false,
    "RangeError": false,
    "ReferenceError": false,
    "Reflect": false,
    "RegExp": false,
    "Set": false,
    "String": false,
    "Symbol": false,
    "SyntaxError": false,
    "toLocaleString": false,
    "toString": false,
    "TypeError": false,
    "Uint16Array": false,
    "Uint32Array": false,
    "Uint8Array": false,
    "Uint8ClampedArray": false,
    "undefined": false,
    "unescape": false,
    "URIError": false,
    "valueOf": false,
    "WeakMap": false,
    "WeakSet": false,
    "arguments": false,
    "GLOBAL": false,
    "root": false,
    "__dirname": false,
    "__filename": false,
    "Buffer": false,
    "clearImmediate": false,
    "clearInterval": false,
    "clearTimeout": false,
    "console": false,
    "exports": true,
    "global": false,
    "Intl": false,
    "module": false,
    "process": false,
    "require": false,
    "setImmediate": false,
    "setInterval": false,
    "setTimeout": false
  },
  "env": {
    "es6": true,
    "node": true
  },
  "rules": {
    "accessor-pairs": "off",
    "array-bracket-newline": "off",
    "array-bracket-spacing": "off",
    "array-callback-return": "off",
    "array-element-newline": "off",
    "arrow-body-style": "off",
    "arrow-parens": "off",
    "arrow-spacing": "off",
    "block-scoped-var": "off",
    "block-spacing": "off",
    "brace-style": "off",
    "callback-return": "off",
    "camelcase": "off",
    "capitalized-comments": "off",
    "class-methods-use-this": "off",
    "comma-dangle": "off",
    "comma-spacing": "off",
    "comma-style": "off",
    "complexity": "off",
    "computed-property-spacing": "off",
    "consistent-return": "off",
    "consistent-this": "off",
    "constructor-super": "error",
    "curly": "off",
    "default-case": "off",
    "dot-location": "off",
    "dot-notation": "off",
    "eol-last": "off",
    "eqeqeq": "off",
    "for-direction": "off",
    "func-call-spacing": "off",
    "func-name-matching": "off",
    "func-names": "off",
    "func-style": "off",
    "function-paren-newline": "off",
    "generator-star-spacing": "off",
    "getter-return": "off",
    "global-require": "off",
    "guard-for-in": "off",
    "handle-callback-err": "off",
    "id-blacklist": "off",
    "id-length": "off",
    "id-match": "off",
    "implicit-arrow-linebreak": "off",
    "indent": [
      "warn",
      4
    ],
    "indent-legacy": "off",
    "init-declarations": "off",
    "jsx-quotes": "off",
    "key-spacing": "off",
    "keyword-spacing": "off",
    "line-comment-position": "off",
    "linebreak-style": [
      "error",
      "unix"
    ],
    "lines-around-comment": "off",
    "lines-around-directive": "off",
    "lines-between-class-members": "off",
    "max-depth": "off",
    "max-len": "off",
    "max-lines": "off",
    "max-nested-callbacks": "off",
    "max-params": "off",
    "max-statements": "off",
    "max-statements-per-line": "off",
    "multiline-comment-style": "off",
    "multiline-ternary": "off",
    "new-cap": "off",
    "new-parens": "off",
    "newline-after-var": "off",
    "newline-before-return": "off",
    "newline-per-chained-call": "off",
    "no-alert": "off",
    "no-array-constructor": "off",
    "no-await-in-loop": "off",
    "no-bitwise": "off",
    "no-buffer-constructor": "off",
    "no-caller": "off",
    "no-case-declarations": "error",
    "no-catch-shadow": "off",
    "no-class-assign": "error",
    "no-compare-neg-zero": "error",
    "no-cond-assign": "error",
    "no-confusing-arrow": "off",
    "no-console": "error",
    "no-const-assign": "error",
    "no-constant-condition": "error",
    "no-continue": "off",
    "no-control-regex": "error",
    "no-debugger": "error",
    "no-delete-var": "error",
    "no-div-regex": "off",
    "no-dupe-args": "error",
    "no-dupe-class-members": "error",
    "no-dupe-keys": "error",
    "no-duplicate-case": "error",
    "no-duplicate-imports": "off",
    "no-else-return": "off",
    "no-empty": "error",
    "no-empty-character-class": "error",
    "no-empty-function": "off",
    "no-empty-pattern": "error",
    "no-eq-null": "off",
    "no-eval": "off",
    "no-ex-assign": "error",
    "no-extend-native": "off",
    "no-extra-bind": "off",
    "no-extra-boolean-cast": "error",
    "no-extra-label": "off",
    "no-extra-parens": "off",
    "no-extra-semi": "error",
    "no-fallthrough": "error",
    "no-floating-decimal": "off",
    "no-func-assign": "error",
    "no-global-assign": "error",
    "no-implicit-coercion": "off",
    "no-implicit-globals": "off",
    "no-implied-eval": "off",
    "no-inline-comments": "off",
    "no-inner-declarations": "error",
    "no-invalid-regexp": "error",
    "no-invalid-this": "off",
    "no-irregular-whitespace": "error",
    "no-iterator": "off",
    "no-label-var": "off",
    "no-labels": "off",
    "no-lone-blocks": "off",
    "no-lonely-if": "off",
    "no-loop-func": "off",
    "no-magic-numbers": "off",
    "no-mixed-operators": "off",
    "no-mixed-requires": "off",
    "no-mixed-spaces-and-tabs": "error",
    "no-multi-assign": "off",
    "no-multi-spaces": "off",
    "no-multi-str": "off",
    "no-multiple-empty-lines": "off",
    "no-native-reassign": "off",
    "no-negated-condition": "off",
    "no-negated-in-lhs": "off",
    "no-nested-ternary": "off",
    "no-new": "off",
    "no-new-func": "off",
    "no-new-object": "off",
    "no-new-require": "off",
    "no-new-symbol": "error",
    "no-new-wrappers": "off",
    "no-obj-calls": "error",
    "no-octal": "error",
    "no-octal-escape": "off",
    "no-param-reassign": "off",
    "no-path-concat": "off",
    "no-plusplus": "off",
    "no-process-env": "off",
    "no-process-exit": "off",
    "no-proto": "off",
    "no-prototype-builtins": "off",
    "no-redeclare": "error",
    "no-regex-spaces": "error",
    "no-restricted-globals": "off",
    "no-restricted-imports": "off",
    "no-restricted-modules": "off",
    "no-restricted-properties": "off",
    "no-restricted-syntax": "off",
    "no-return-assign": "off",
    "no-return-await": "off",
    "no-script-url": "off",
    "no-self-assign": "error",
    "no-self-compare": "off",
    "no-sequences": "off",
    "no-shadow": "off",
    "no-shadow-restricted-names": "off",
    "no-spaced-func": "off",
    "no-sparse-arrays": "error",
    "no-sync": "off",
    "no-tabs": "off",
    "no-template-curly-in-string": "off",
    "no-ternary": "off",
    "no-this-before-super": "error",
    "no-throw-literal": "off",
    "no-trailing-spaces": "off",
    "no-undef": "error",
    "no-undef-init": "off",
    "no-undefined": "off",
    "no-underscore-dangle": "off",
    "no-unexpected-multiline": "error",
    "no-unmodified-loop-condition": "off",
    "no-unneeded-ternary": "off",
    "no-unreachable": "error",
    "no-unsafe-finally": "error",
    "no-unsafe-negation": "error",
    "no-unused-expressions": "off",
    "no-unused-labels": "error",
    "no-unused-vars": "error",
    "no-use-before-define": "off",
    "no-useless-call": "off",
    "no-useless-computed-key": "off",
    "no-useless-concat": "off",
    "no-useless-constructor": "off",
    "no-useless-escape": "error",
    "no-useless-rename": "off",
    "no-useless-return": "off",
    "no-var": "off",
    "no-void": "off",
    "no-warning-comments": "off",
    "no-whitespace-before-property": "off",
    "no-with": "off",
    "nonblock-statement-body-position": "off",
    "object-curly-newline": "off",
    "object-curly-spacing": "off",
    "object-property-newline": "off",
    "object-shorthand": "off",
    "one-var": "off",
    "one-var-declaration-per-line": "off",
    "operator-assignment": "off",
    "operator-linebreak": "off",
    "padded-blocks": "off",
    "padding-line-between-statements": "off",
    "prefer-arrow-callback": "off",
    "prefer-const": "off",
    "prefer-destructuring": "off",
    "prefer-numeric-literals": "off",
    "prefer-promise-reject-errors": "off",
    "prefer-reflect": "off",
    "prefer-rest-params": "off",
    "prefer-spread": "off",
    "prefer-template": "off",
    "quote-props": "off",
    "quotes": [
      "error",
      "single"
    ],
    "radix": "off",
    "require-await": "warn",
    "require-jsdoc": "off",
    "require-yield": "error",
    "rest-spread-spacing": "off",
    "semi": [
      "error",
      "always"
    ],
    "semi-spacing": "off",
    "semi-style": "off",
    "sort-imports": "off",
    "sort-keys": "off",
    "sort-vars": "off",
    "space-before-blocks": "off",
    "space-before-function-paren": "off",
    "space-in-parens": "off",
    "space-infix-ops": "off",
    "space-unary-ops": "off",
    "spaced-comment": "off",
    "strict": "off",
    "switch-colon-spacing": "off",
    "symbol-description": "off",
    "template-curly-spacing": "off",
    "template-tag-spacing": "off",
    "unicode-bom": "off",
    "use-isnan": "error",
    "valid-jsdoc": "off",
    "valid-typeof": "error",
    "vars-on-top": "off",
    "wrap-iife": "off",
    "wrap-regex": "off",
    "yield-star-spacing": "off",
    "yoda": "off"
  },
  "parserOptions": {
    "ecmaVersion": 6,
    "ecmaFeatures": {
      "globalReturn": true,
      "ecmaVersion": 8,
      "experimentalObjectRestSpread": true
    }
  },
  "extends": "eslint:recommended"
}
```

**Comment by not-an-aardvark** [comment]
@codeblooded I think your error is unrelated. In the `parserOptions` section, you have:

```
  "parserOptions": {
    "ecmaVersion": 6,
    "ecmaFeatures": {
      "globalReturn": true,
      "ecmaVersion": 8,
      "experimentalObjectRestSpread": true
    }
  },
```

This should just be:

```json
  "parserOptions": {
    "ecmaVersion": 8,
    "ecmaFeatures": {
      "globalReturn": true,
      "experimentalObjectRestSpread": true
    }
  },
```

`ecmaVersion` generally isn't supposed to go in the `ecmaFeatures` object.

**Comment by bittib** [comment]
I have the same problem. 

const foo = async (event, context, callback) {}
Get error:
error  Parsing error: Unexpected token {

./node_modules/.bin/eslint -v
v4.18.2

{
    "env": {
        "browser": true,
        "commonjs": true,
        "es6": true,
        "node": true
    },
    "extends": "eslint:recommended",
    "parserOptions": {
        "ecmaVersion": 8,
        "ecmaFeatures": {
            "globalReturn": true,
            "experimentalObjectRestSpread": true
        }
    },
    "rules": {
        "block-scoped-var": "error",
        "comma-dangle": ["error", "always-multiline"],
        "curly": ["error", "all"],
        "dot-notation": "error",
        "eqeqeq": "error",
        "no-alert": "error",
        "no-caller": "error",
        "no-console": ["off"],
        "no-div-regex": "error",
        "no-eval": "error",
        "no-extend-native": "error",
        "no-extra-bind": "error",
        "no-floating-decimal": "error",
        "no-implicit-coercion": "error",
        "no-implied-eval": "error",
        "no-lone-blocks": "error",
        "no-loop-func": "error",
        "no-multi-spaces": "error",
        "no-native-reassign": "error",
        "no-new-func": "error",
        "no-return-assign": "error",
        "no-sequences": "error",
        "no-unused-expressions": "error",
        "no-unused-vars": ["error", {"argsIgnorePattern": "^_" }],
        "no-useless-call": "error",
        "no-void": "error",
        "radix": "error",
        "wrap-iife": "error",
        "no-use-before-define": ["error", {"functions": false, "classes": false, "variables": false}],
        "callback-return": "error",
        "handle-callback-err": "error",
        "brace-style": ["error", "1tbs", { "allowSingleLine": true }],
        "camelcase": ["error", {"properties": "never"}],
        "comma-spacing": ["error", {"after":true}],
        "comma-style": "error",
        "eol-last": "error",
        "indent": ["error", 4],
        "key-spacing": ["error", {"afterColon": true}],
        "new-cap": ["error", {"newIsCap": true}],
        "new-parens": "error",
        "no-array-constructor": "error",
        "no-multiple-empty-lines": ["error", {"max" :1}],
        "no-nested-ternary": "error",
        "no-new-object": "error",
        "no-trailing-spaces": "error",
        "padded-blocks": ["error", {"blocks": "never", "switches": "never", "classes": "never"}],
        "quotes": ["error", "single", {"avoidEscape": true, "allowTemplateLiterals": true}],
        "quote-props": ["error", "as-needed"],
        "semi": ["error", "never"],
        "sort-vars": ["error", {"ignoreCase": false}],
        "space-before-blocks": ["error", "always"],
        "space-before-function-paren": "error",
        "space-infix-ops": "error",
        "space-unary-ops": ["error", {"words": true}],
        "spaced-comment": ["error", "always", {
            "line": {"markers": ["*package", "!", "/", ",", "="]},
            "block": {"balanced": true, "markers": ["*package", "!", ",", ":", "::", "flow-include"], "exceptions": ["*"]}
        }],
        "arrow-spacing": ["error", {"before": true, "after": true}],
        "no-class-assign": "error",
        "no-const-assign": "error",
        "no-this-before-super": "error",
        "no-var": "error",
        "prefer-const": "error",
        "prefer-spread": "error"
    }
}

**Comment by Koviac** [comment]
@not-an-aardvark 
Node v8.9.4
NPM v6.4.0
Eslint v5.4.0 

Eslint flags async/await functions or statements saying:  'Async functions are not supported until Node.js 7.6.0. The configured version range is '>=6.0.0'. (node/no-unsupported-features/es-syntax)'

.eslintrc.json
{
  "env": {
      "browser": true,
      "es6": true,
      "node": true
  },
  "extends": ["airbnb-base", "plugin:angular/johnpapa", "plugin:node/recommended"],
  "parserOptions": {
    "ecmaVersion": 2017
  },
}

Please what could I possibly be doing wrong

**Comment by platinumazure** [comment]
@Koviac Your issue is unrelated-- eslint-plugin-node is giving you a lint message. In your case, you need to check your "engines" field in package.json-- you can't use arrow functions in Node engines before 7.6.0. If you have further questions, please open an issue in eslint-plugin-node.

**Comment by Koviac** [comment]
@platinumazure 
thanks for the reply. is there a difference between Node Engines and Node Version? If no, why the flag.

**Comment by platinumazure** [comment]
@Koviac Yes, ecmaVersion and Node engine/version are completely different:

* ecmaVersion determines what version of ECMAScript ESLint will be able to parse and understand. If you don't set this correctly, you get parsing errors where ESLint can't even lint a file due to not understanding a particular token.
* The `engines` field in package.json represents what version of Node.js your package is compatible with. Node.js implements language features separately and might not support all of ES2017 (or whichever certain of the language) in all versions. For example, async function support was introduced in Node 7.6.

The eslint-plugin-node plugin is looking at your declared supported Node versions (6.0 or greater) and identifying that you are using async functions (only work on Node 7.6.0 or greater). So anyone who uses your package while running Node 6.x will get a runtime error. This is showing up as a lint error, not a parser error. ESLint is reading the async function fine, but eslint-plugin-node is warning you that your Node 6.x users will have trouble with your package.

If you have any further questions, please stop by our [Gitter chat](https://gitter.im/eslint/eslint) or the eslint-plugin-node repository. Thanks!

**Comment by nzakas** [comment]
It looks like the conversation is stalled here. As this is a question rather than an action item, I'm closing the issue. If you still need help, please send a message to our [mailing list](https://groups.google.com/group/eslint) or [chatroom](https://gitter.im/eslint/eslint). Thanks!

**Comment by uclaeagit** [comment]
FWIW here's the .eslintrc file I finally arrived at to fix this problem:

```
{
  "parserOptions": {
    "ecmaVersion": 6,
    "sourceType": "module",
    "ecmaFeatures": {
      "experimentalObjectRestSpread": true
    },
  },

  "env": {
    "es6": true,
    "node": true
  },
  
  "extends": "eslint:recommended",

}
```
If I remove either the experimentalObjectRestSpread entry or the "extends": "eslint:recommended" line - it breaks again. And not just like "I don't like this arrow function". The parser crashes and stops when it sees that. Maybe this will help.

Also fun fact I just discovered - if I put the .eslintrc file in the parent directory of all my projects - SublimeLinter will still find it - and I don't have to create/copy a new one into each project. I create a ton of little projects that come and go, so this is going to be a huge help for me.
